lib/ansible/modules/cloud/vmware/vmware_vcenter_permission.py

```diff
@@ -0,0 +1,359 @@
+          type: 'Datacenter'
+    principal:
+          name: 'test'
+          domain: 'vsphere.local'
+          isGroup: False
+    propagate: True
+    state: present
+
+- name: remove permission from datacenter DC
+  vmware_vcenter_permission:
+    hostname: "{{ vCenter_hostname }}"
+    username: "{{ vCenter_username }}"
+    password: "{{ vCenter_password }}"
+    validate_certs: False
+    role_name: 'VM-Power'
+    entity:
+          name: 'DC'
+          type: 'Datacenter'
+    principal:
+          name: 'test'
+          domain: 'vsphere.local'
+          isGroup: False
+    propagate: True
+    state: absent
+'''
+
+RETURN = '''
+
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import (HAS_PYVMOMI,
+                                         connect_to_api,
+                                         vmware_argument_spec,
+                                         find_object_by_name
+                                         )
+
+class VmwareVcenterPermission(object):
+
+    def __init__(self, module):
+        self.module = module
+        self.state = self.module.params['state']
+
+        self.content = connect_to_api(module)
+        self.authMgr = self.content.authorizationManager
+
+        if self.state == 'present':
+            self.role_name = self.module.params['role_name']
+            self.role      = self.get_role(self.role_name)
+        else:
+            self.role_name = None
+            self.role      = None
+
+        self.propagate = self.module.params['propagate']
+
+        self.entity = self.get_entity(self.module.params['entity'])
+        self.principal = self.module.params['principal']
+
+    def process_state(self):
+        """map current vs desired state and change accordingly"""
+        try:
+            role_states = {
+                'absent': {
+                    'present': self.state_delete_permission,
+                    'absent': self.state_exit_unchanged,
+                },
+                'present': {
+                    'update': self.state_update_permission,
+                    'present': self.state_exit_unchanged,
+                    'absent': self.state_set_permission,
+                }
+            }
+            role_states[self.state][self.check_permission()]()
+        except vmodl.RuntimeFault as runtime_fault:
+            self.module.fail_json(msg=runtime_fault.msg)
+        except vmodl.MethodFault as method_fault:
+            self.module.fail_json(msg=method_fault.msg)
+        except Exception as e:
+            self.module.fail_json(msg=str(e))
+
+    def check_permission(self):
+        """checks if desired state matches actual state"""
+        perms = self.authMgr.RetrieveEntityPermissions(self.entity, False)
+        for perm in perms:
+            # check for a permission assigned to the principal
```

Comment:
The condition is a bit long, I would write it this way instead:

```python
if perm.principal != self.principal['domain'].upper() + '\\' + self.principal['name']:
    continue
if perm.group != self.principal['isGroup']:
    continue
```

This way, you avoid an extra level of indent.